shared/js/toc.jsx

```diff
@@ -0,0 +1,772 @@
+            isDesktop={this.props.isDesktop} />
+        </div>
+      );
+    }
+  });
+
+  var RoomPresenceView = React.createClass({
+    propTypes: {},
+
+    render: function() {
+      return (
+        <div className="room-active-users">
+          <div className="room-user" data-name="Pau Masiá">
+            <span>{'P'}</span>
+          </div>
+          <div className="room-user" data-name="Manu">
+            <span>{'M'}</span>
+          </div>
+        </div>
+      );
+    }
+  });
+
+  var RoomActionsView = React.createClass({
+    propTypes: {
+      addUrlTile: React.PropTypes.func.isRequired,
+      isDesktop: React.PropTypes.bool.isRequired
+    },
+
+    getInitialState: function() {
+      return {
+        showAddUrlPanel: false
+      };
+    },
+
+    toggleAddUrlPanel: function() {
+      this.setState({
+        showAddUrlPanel: !this.state.showAddUrlPanel
+      });
+    },
+
+    handleAddUrlClick: function(url) {
+      this.toggleAddUrlPanel();
+      this.props.addUrlTile(url);
+    },
+
+    render: function() {
+      return (
+        <div className="room-actions-buttons">
+          <div className="room-action-add-url">
+            <button className="add-url" onClick={this.toggleAddUrlPanel} />
+            {
+              this.state.showAddUrlPanel ?
+                <AddUrlPanelView
+                  handleAddUrlClick={this.handleAddUrlClick} /> : null
+            }
+          </div>
+          <div className="room-action-settings">
+            <button className="settings" />
+          </div>
+        </div>
+      );
+    }
+  });
+
+  var AddUrlPanelView = React.createClass({
+    propTypes: {
+      handleAddUrlClick: React.PropTypes.func.isRequired
+    },
+
+    handleClick: function(event) {
+      event.preventDefault();
+      var input = this.refs.siteUrl.getDOMNode();
+      input.value && this.props.handleAddUrlClick(input.value);
+    },
+
+    render: function() {
+      return (
+        <div className="room-panel-add-url">
+          <h2>{'Add a site to the room'}</h2>
+          <input placeholder="http://..." ref="siteUrl" type="text" />
+          <button onClick={this.handleClick}>{'Add site'}</button>
+        </div>
+      );
+    }
+  });
+
+  var RoomContentView = React.createClass({
+    propTypes: {
+      tiles: React.PropTypes.array
+    },
+
+    getDefaultProps: function() {
+      return {
+        tiles: []
+      };
+    },
+
+    render: function() {
+      return (
+        <div className="room-toc">
+          {
+            this.props.tiles.map(function(tile, index) {
+              return (
+                <TileView
+                  key={index}
+                  tile={tile} />
+              );
+            }, this)
+          }
+        </div>
+      );
+    }
+  });
+
+  var TileView = React.createClass({
+    statics: {
+      GOOGLE_SCREENSHOT: "https://www.googleapis.com/pagespeedonline/v1/runPagespeed?screenshot=true&strategy=desktop&url="
```

Comment:
We're not going to be able to use this without a review. I know we've talked about implementing our own version for the non-desktop side. I think we should probably land this patch without this and add in something better later. Although its a nice demo, it sets potentially wrong expectations, and we don't actually need it for development purposes.